File: Data Structures & Algorithms/remove-element/submission-1.java
class Solution {
    public int removeElement(int[] nums, int val) {
       //brute force way - extra array
        //Optimized solution: just do a one pass and track the number of elements that not equal to val
        int len = nums.length;
        int k = 0;

        for (int i = 0; i < len; i++) {
            if(nums[i] != val){
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # use a hashmap
        hashMap = {}
        length = len(nums)

        for i in range(0, length):
            hashMap[nums[i]] = nums[i] 
            
        if length > len(hashMap):
            return True
        else:
            return False
         